brinqa_connector.py

```diff
@@ -0,0 +1,459 @@
+        user_post_body = {
+            "query": "query confirmGraphqlAccessible {assets(limit: 1) {display}}"
+        }
+        ret_val, response = self._make_rest_call(
+            "/graphql/tvm",
+            action_result,
+            headers=headers,
+            json=user_post_body,
+            method="post",
+        )
+        if phantom.is_fail(ret_val):
+            self.debug_print("Test Connectivity Failed: Graph Response")
+            return RetVal(
+                action_result.set_status(
+                    phantom.APP_ERROR,
+                    status_message="Test Connectivity Failed: Graph Response. Check to make sure the credentials are correct.",
+                ),
+                None,
+            )
+        else:
+            if response["data"]["assets"]:
+                self.debug_print("Test Connectivity Passed")
+                return RetVal(
+                    action_result.set_status(
+                        phantom.APP_SUCCESS,
+                        status_message="Test Connectivity Passed",
+                    ),
+                    None,
+                )
+            else:
+                self.debug_print(
+                    "Test Connectivity Failed: Data. Check to make sure the account has GraphQL Permissions."
+                )
+                return RetVal(
+                    action_result.set_status(
+                        phantom.APP_ERROR,
+                        status_message="Test Connectivity Failed: Data. Check to make sure the account has GraphQL Permissions.",
+                    ),
+                    None,
+                )
+
+    def _handle_query_brinqa(self, param) -> str:
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        self.debug_print("Logging in")
+        self._make_rest_call(
+            f"/api/auth/authMethod?email={self._username}", action_result, method="get"
+        )
+        # make an auth rest call to login to Brinqa
+        auth_post_body = {"username": self._username, "password": self._password}
+        ret_val, response = self._make_rest_call(
+            "/api/auth/login", action_result, json=auth_post_body, method="post"
+        )
+        if phantom.is_fail(ret_val):
+            self.debug_print("Failed login.")
+            return action_result.get_status()
+        else:
+            token = response["access_token"]
+            token_type = response["token_type"]
+
+        self.debug_print("Connecting to endpoint")
+        # make a graph rest call to receive information about all items in identifier
+        headers = {"Authorization": token_type + " " + token}
+        if "filter" in param and "return_values" in param:
+            data_model = f"{param['data_model']}"
+            data_model_lower = data_model[0].lower() + data_model[1:]
+            filter_string = f"{param['filter']}"
+            return_string = f"{{{param['return_values']}}}"
```

Comment:
What is the reasoning for additional brackets here?

Reply:
The 3 brackets would create a string surrounded with {}. @Jth155, Can you please confirm whether it is the actual use-case or not?

Reply:
Hi @Jth155 , Can you please take a look into this?

Reply:
Yes, that is the intended use case. The return values are a bracketed set of attributes that are space delineated at the end of the query.

Reply:
@alexa-phantom , Can you please re-review this?